expression/builtin_info.go

```diff
@@ -384,7 +391,212 @@ type benchmarkFunctionClass struct {
 }
 
 func (c *benchmarkFunctionClass) getFunction(ctx sessionctx.Context, args []Expression) (builtinFunc, error) {
-	return nil, errFunctionNotExists.GenWithStackByArgs("FUNCTION", "BENCHMARK")
+	if err := c.verifyArgs(args); err != nil {
+		return nil, errors.Trace(err)
+	}
+	argFieldTp := args[1].GetType()
+	argTp := argFieldTp.EvalType()
+
+	bf := newBaseBuiltinFuncWithTp(ctx, args, types.ETInt, types.ETInt, argTp)
+	var sig builtinFunc
+	switch argTp {
+	case types.ETString:
+		sig = &builtinBenchmarkStringSig{bf}
+	case types.ETInt:
+		sig = &builtinBenchmarkIntSig{bf}
+	case types.ETJson:
+		sig = &builtinBenchmarkJSONSig{bf}
+	case types.ETReal:
+		sig = &builtinBenchmarkRealSig{bf}
+	case types.ETDecimal:
+		sig = &builtinBenchmarkDecimalSig{bf}
+	case types.ETDuration:
+		sig = &builtinBenchmarkIntSig{bf}
+	case types.ETDatetime:
+		sig = &builtinBenchmarkTimeSig{bf}
+	case types.ETTimestamp:
+		sig = &builtinBenchmarkTimeSig{bf}
+	default:
+		return nil, errors.New("FUNCTION benchmark has wrong arg2")
+	}
+
+	return sig, nil
+
+}
+
+type builtinBenchmarkStringSig struct {
+	baseBuiltinFunc
+}
+
+func (b *builtinBenchmarkStringSig) evalInt(row chunk.Row) (int64, bool, error) {
+	x, isNull, err := b.args[0].EvalInt(b.ctx, row)
+	if isNull || err != nil {
+		return 0, isNull, errors.Trace(err)
+	}
+	var i int64 = 0
+	for ; i < x; i++ {
+		_, _, err := b.args[1].EvalString(b.ctx, row)
+		if err != nil {
+			return 0, isNull, errors.Trace(err)
```

Comment:
`return 0, true, errors.Trace(err)`, you may need to change the other places in this patch as well.